src-js/instantsearch.js

```diff
@@ -0,0 +1,76 @@
+import * as throttle from 'lodash.throttle';
+
+// Instant search initialization
+export function init () {
+  var search = instantsearch({
+    appId: 'U0RXNGRK45',
+    apiKey: 'dd49f5b81d238d474b49645a4daed322', // search-only API Key; safe for front-end code
+    indexName: 'documentation',
+    routing: true
+  });
+
+  // initialize SearchBox
+  search.addWidget(
+    instantsearch.widgets.searchBox({
+      container: '#search-box',
+      cssClasses: {
+        input: 'instantsearch-search'
+      },
+      placeholder: 'Search Documentation',
+      queryHook: throttle(function (query, searchFunction) { searchFunction(query); }, 1000) // method to throttle search requests
+    })
+  );
+
+  // initialize SearchBox
+  search.addWidget(
+    instantsearch.widgets.searchBox({
+      container: '#mobile-search-box',
+      cssClasses: {
+        input: 'instantsearch-search'
+      },
+      placeholder: 'Search Documentation'
+    })
+  );
```

Comment:
Do we need to have two separate `searchBox`es?

Reply:
yeah, due to positioning in the dom, I think it's the simplest solution -- but let me know if you want to sync, or think there's a better way.

Reply:
@trevor-circleci Could you explain a little more what you mean by positioning in the dom?

This is what I'm seeing at <http://localhost:4000/docs>, and they look like they're directly adjacent siblings 😀

![image](https://user-images.githubusercontent.com/4341496/46307836-ab47e680-c57d-11e8-92e9-cedec1464c7a.png)

Reply:
@trevor-circleci Could you explain a little more what you mean by positioning in the dom?

From inspecting those elements, all I could see is that we have some styles on `.main-searchbar` that we don't want to be nested under?

Reply:
I'm concerned about this as well 😅 For one, I think we're rendering and instantiating two separate searchboxes. This is apparent if you apply this diff
```diff
diff --git a/jekyll/_sass/_instantsearch.scss b/jekyll/_sass/_instantsearch.scss
index 2a283247..ea4275a6 100644
--- a/jekyll/_sass/_instantsearch.scss
+++ b/jekyll/_sass/_instantsearch.scss
@@ -8,7 +8,7 @@
 .main-searchbar {
   border-top: 1px solid $gray-200;
   border-bottom: 1px solid $gray-200;
-  position: fixed;
+  /* position: fixed; */
   width: 100%;
   background: white;
   z-index: 2;
```

![docs-search](https://user-images.githubusercontent.com/4341496/46309400-daf8ed80-c581-11e8-9a0f-14711d82a94a.gif)